### DISCUSSION TITLE
mpv doesn't exit

### DISCUSSION THREAD

**Discussion by angousana** [mpv-player/mpv#14945] [Q&A] [answered]
with lua if i run os.execute('sleep 10000') in background without blocking the video I can't exit with q or ctrl + c, even if i try to bind k with os.execute('pkill -9 mpv') it waits for the sleep command to finish, is there any way i can start a new thread or something?

**Comment by guidocella** [comment]
Pressing ctrl+c twice should quit.

**Comment by angousana** [comment]
Yes that would work, but the sleep command was like a metaphorical command,
i made an auto-sub script and it downloads all available subs at once but
sometimes i start the wrong episode and when i click q or ctrl+c mpv
freezes until all the subs are downloaded, ctrl+c closes the os.execute but
not the whole function so it just downloads the next sub in queue, is there
maybe a way to check if I've pressed ctrl+c?, i tried binding a key to just
kill -9 the whole mpv but that doesn't freeze mpv it just waits for the
other functions to finish and is executed after all the downloading is done

**Reply by guidocella** [reply, marked as answer]
You can place this in a different script:

```lua
mp.register_event('shutdown', function ()
    os.exit()
end)
```

**Reply by angousana** [reply]
Thanks that worked!

**Reply by angousana** [reply]
mpv still hangs when i try to play the next/prev file in the playlist

**Reply by angousana** [reply]
this worked
```lua
for item in findSubs() do
    if mp.get_property_bool('user-data/autosub/abandon_old_instance') then
        return mp.set_property_bool('user-data/autosub/abandon_old_instance', false)
    end
    
    ... --Download subs
end

-- in another script file
mp.add_hook('on_unload', 0, function() mp.set_property_bool('user-data/autosub/abandon_old_instance', true) end)
mp.add_hook('on_preloaded', 0, function()
    if mp.get_property_bool('user-data/autosub/abandon_old_instance') then
        mp.set_property_bool('user-data/autosub/abandon_old_instance', false)
    end
end)

```
